src/GraphQL/Execution/DocumentExecuter.cs

```diff
@@ -292,8 +293,8 @@ protected virtual ExecutionContext BuildExecutionContext(ExecutionOptions option
 
             Operation = operation,
             Variables = validationResult.Variables ?? Variables.None,
-            ArgumentValues = validationResult.ArgumentValues,
-            DirectiveValues = validationResult.DirectiveValues,
+            ArgumentValues = validationResult.ArgumentValues == null ? null : new ConcurrentDictionary<GraphQLField, IDictionary<string, ArgumentValue>>(validationResult.ArgumentValues),
+            DirectiveValues = validationResult.DirectiveValues == null ? null : new ConcurrentDictionary<ASTNode, IDictionary<string, DirectiveInfo>>(validationResult.DirectiveValues),
```

Comment:
For merged selection sets, GraphQL.NET creates a faux `GraphQLField` containing the merged selection set as the only change.  While one solution would be to overwrite the `SelectionSet` property of the first `GraphQLField`, this will change the original representation of the query and would mess up document caching at the very least, so this is not possible.  So we are updating the dictionaries containing the arguments and directives to point to the original field arguments.  Unfortunately this occurs at runtime, and possibly within a parallel execution strategy.  So a `ConcurrentDictionary` is used.  We cannot create a `ConcurrentDictionary` at inception because the `ValidationContext` is hard typed to a `Dictionary<TKey, TValue>`.  So we end up copying the dictionary within the document executer into a concurrent dictionary.